File: Problems/No.949_Largest_Time_for_Given_Digits/largest_time_for_given_digits.py
class Solution(object):
    def largestTimeFromDigits(self, arr):
        """
        :type arr: List[int]
        :rtype: str
        """

        # parameter
        valid_time = []

        # start computing
        for d0, d1, d2, d3 in list(itertools.permutations(arr, 4)):
            hrs = d0*10 + d1
            if ((hrs < 24) and (d2 < 6)):
                valid_time.append([hrs, d2*10 + d3])
        
        if not valid_time:
            return ""
        else:
            hrs, mins = max(valid_time)
            return str(hrs).rjust(2, '0') + ":" + str(mins).rjust(2, '0')

File: Problems/No.949_Largest_Time_for_Given_Digits/largest_time_for_given_digits.cpp
class Solution {
public:
    string largestTimeFromDigits(vector<int>& arr) {
        
        // parameter
        std::string latest_time = "";
        std::vector<int> temp_time;

        // start computing
        std::sort(arr.begin(), arr.end());
        do {

            int hr = arr[0]*10 + arr[1];
            if ((hr < 24) && (arr[2] < 6)) temp_time = arr;
            else if (hr >= 24) break;
            else continue;

        } while (std::next_permutation(arr.begin(), arr.end()));

        if (temp_time.empty()) return "";
        else {

            latest_time = std::to_string(temp_time[0]) + std::to_string(temp_time[1]) + ":"
                        + std::to_string(temp_time[2]) + std::to_string(temp_time[3]);
            
            return latest_time;

        }

    }
};
